### ISSUE TITLE
[Feature Request] add UKSM as optional patch.

### ISSUE THREAD

**Issue by dreirund** [Frogging-Family/linux-tkg#780]
As a feature request, I want to request to add [&nearr;&hairsp;uksm](https://pfkernel.natalenko.name/) to be available as optional patch.

And this then needs the following build logic:  
When uksm is patched in, the `PKGBUILD`'s `provides` array should be enhanced with `UKSMD-BUILTIN` and `optdepends` should be enhanced with `uksmd:  Userspace KSM helper daemon` or alike.

See the `-pf`-kernels, which have uksm.

Regards!

**Comment by dreirund** [comment]
> We used to offer it but it gave negative results in gaming so it was
> dropped. I guess we could add it to community patches for people
> still wanting it in.

Ah OK!

As community patch would be nice.

Since you dropped it -- where is the latest one you had? (I am not
familiar with `git` to be able to correctly search the git hostory.)

Regards!

**Comment by ptr1337** [comment]
> We used to offer it but it gave negative results in gaming so it was dropped. I guess we could add it to community patches for people still wanting it in.

I think you are mixing it up with "ultra ksm" patches.

**Comment by dreirund** [comment]
> > We used to offer it but it gave negative results in gaming so it
> > was dropped. I guess we could add it to community patches for
> > people still wanting it in.  
> 
> I think you are mixing it up with "ultra ksm" patches.

"ultra ksm" is dead I think (is it not?). I mean "userspace ksm", which
is part of the pfkernel.

**Comment by ptr1337** [comment]
> > We used to offer it but it gave negative results > in gaming so it was dropped. I guess we could add > it to community patches for people still wanting it in. I think you are mixing it up with "ultra ksm" patches.
> "ultra ksm" is dead I think (is it not?). I mean "userspace ksm", which is part of the pf-kernel [1] [1] https://pfkernel.natalenko.name/

Yes, correct. The gaming related issue came once from "ultra ksm" and not "userspace ksm daemon". Just wanted to point that out. :)

**Comment by Tk-Glitch** [comment]
Oh, indeed. We can definitely check that out then.

**Comment by mabod** [comment]
Just out of curiosity: What is this good for? I was reading this page: https://codeberg.org/pf-kernel/uksmd
but do not understand the benefits.

**Comment by micros24** [comment]
> Just out of curiosity: What is this good for? I was reading this page: https://codeberg.org/pf-kernel/uksmd
> but do not understand the benefits.
> 

It's a memory deduplication daemon. Basically, it's helps save RAM space from programs that have been running for a period of time.

**Comment by sirlucjan** [comment]
It is worth adding, however, that kernel patches are useless unless you provide an additional uksmd package with the uksmd daemon.

**Comment by dreirund** [comment]
> It is worth adding, however, that kernel patches are useless unless
> you provide an additional uksmd package with the uksmd daemon. 

It is available:

https://codeberg.org/pf-kernel/uksmd

https://aur.archlinux.org/packages/uksmd
https://aur.archlinux.org/packages/uksmd-git
https://aur.archlinux.org/packages/uksmd-nosystemd-git

**Comment by sirlucjan** [comment]
In CachyOS  we provide a special version of UKSMD - we added uksmdstats, which shows accurate statistics. 

```
lucjan at cachyos ~ 12:34:00    
❯ uksmdstats   
======================================================
uksmd with uksmdstats support. Powered by CachyOS Team
======================================================
Full scans               18       
Interval                 20 ms    
                                  
Max page sharing ratio   256      
Pages to scan            100      
Pages over ratio         3        
Duplicated pages         143      
Use zero pages           0        
Sharing/shared ratio     30.7024  
Unshared/sharing ratio   4.36666  
                                  
Pages sharing            164.426  MiB
Pages shared             5.35547  MiB
Pages unshared           717.992  MiB
                                  
General profit           140.06   MiB
```

Polish and Russian translations are also available. 

```
ucjan at cachyos ~ 12:33:41    
❯ uksmdstats
======================================================
uksmd with uksmdstats support. Powered by CachyOS Team
======================================================
Pełne skany                         18       
Interwał                            20 ms    
                                             
Maks. wskaźnik stronicowania        256      
Strony do skanowania                100      
Strony ponad wskaźnik               2        
Zdublowane strony                   138      
Użycie stron zerowych               0        
Wskaźnik stron współdzielonych      29.2094  
Wskaźnik stron niewspółdzielonych   4.08037  
                                             
Strony dost. do współdzielenia      161.793  MiB
Strony współdzielone                5.53906  MiB
Strony niewspółdzielone             660.176  MiB
                                             
Ogólny zysk                         137.403  MiB
lucjan at cachyos ~ 12:33:43    
```
https://github.com/CachyOS/uksmd/
https://github.com/CachyOS/CachyOS-PKGBUILDS/tree/master/uksmd

A version for Fedora is also available:

https://copr.fedorainfracloud.org/coprs/bieszczaders/kernel-cachyos-addons/builds/

However, it should be added that the latest UKSMD requires procps-ng version 4.X

**Comment by dreirund** [comment]
> *In CachyOS we provide a special version of UKSMD*

@sirlucjan, Can you [&nearr;&hairsp;give back your changes to upstream UKSMD](https://codeberg.org/pf-kernel/uksmd/issues/6), please?

Regards!

**Comment by sirlucjan** [comment]
This is a special version for CachyOS, you can use it. It is fully compatible
